src/extension/wps/wps-core/src/main/java/org/geoserver/wps/ppio/GeoTiffPPIO.java

```diff
@@ -65,26 +65,35 @@
 
     private static final Logger LOGGER = Logging.getLogger(GeoTiffPPIO.class);
 
-    protected GeoTiffPPIO() {
+    private final WPSResourceManager resources;
+
+    protected GeoTiffPPIO(WPSResourceManager resources) {
         super(GridCoverage2D.class, GridCoverage2D.class, "image/tiff");
+        this.resources = resources;
     }
 
     @Override
     public Object decode(InputStream input) throws Exception {
         // in order to read a grid coverage we need to first store it on disk
         File root = new File(System.getProperty("java.io.tmpdir", "."));
-        File f = File.createTempFile("wps", "tiff", root);
-        try (FileOutputStream os = new FileOutputStream(f)) {
-            IOUtils.copy(input, os);
-        }
-
-        // and then we try to read it as a geotiff
-        AbstractGridFormat format = GridFormatFinder.findFormat(f);
-        if (format instanceof UnknownFormat) {
-            throw new WPSException(
-                    "Could not find the GeoTIFF GT2 format, please check it's in the classpath");
+        File file = File.createTempFile("wps", ".tiff", root);
+        GridCoverageReaderResource resource = null;
+        try {
+            FileUtils.copyInputStreamToFile(input, file);
+            GeoTiffFormat format = new GeoTiffFormat();
+            if (!format.accepts(file)) {
+                throw new WPSException("Could not read " + getMimeType() + " coverage");
+            }
+            GeoTiffReader reader = format.getReader(file);
```

Comment:
This may be a behavioral change. The old code would have ended up accepting pretty much anything for which a format is registered, but wondering if now the code can still read an improper geotiff, like, tiff without geotags along with tfw and prj files. The old code would have, using the WorldImageReader. @dromagnoli do you remember if the GeoTIFFReader can read the triplet above?

Reply:
The WorldImage format actually doesn't work with the current GeoTIFF PPIO because the WorldImage format checks the file extension to determine whether it can read the image but the current temp file has no file extension.  There is a behavior change if coverage formats from some extensions were being used that do not rely on the file name/extension.  This change also causes another behavior change where it was previously possible for the GeoTIFF PPIO to read ArcGrid files so users wouldn't have to explicitly specify the ArcGrid mimetype in the request.  I will change this section back to the original code.

Reply:
I fixed the GeoTIFF PPIO behavior to be the same as it is now.

Reply:
I fixed the GeoTIFF PPIO behavior to be the same as it is now.